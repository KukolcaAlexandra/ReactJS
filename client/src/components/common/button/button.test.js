import React from 'react';
import Button from './button.component';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Button component', () => {
    
  it('big button should be render correctly', () => {
    const onClick = jest.fn();
    const component = shallow(<Button title="Search" onClick size='big'/>);
    expect(component).toMatchSnapshot();
  });

  it('medium button should be render correctly', () => {
    const onClick = jest.fn();
    const component = shallow(<Button title="Genre" onClick size='medium'/>);
    expect(component).toMatchSnapshot();
  });

  it('small button should be render correctly', () => {
    const onClick = jest.fn();
    const component = shallow(<Button title="Search" onClick size='small'/>);
    expect(component).toMatchSnapshot();
  });

  /*it('should be call onClickHandler', () => {
    const onClick = jest.fn();
    const component = mount(<Button title="Search" onClick={onClick} size='big'/>);
    const button = component.find('[type="button"]').simulate('click');
    expect(onClick).toHaveBeenCalled();
  })*/

})
