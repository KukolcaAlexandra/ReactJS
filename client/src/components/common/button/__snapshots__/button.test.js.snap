// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component big button should be render correctly 1`] = `ShallowWrapper {}`;

exports[`Button component medium button should be render correctly 1`] = `ShallowWrapper {}`;

exports[`Button component small button should be render correctly 1`] = `ShallowWrapper {}`;
